plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/GenerateGramineSGXManifest.kt

```diff
@@ -0,0 +1,89 @@
+package com.r3.conclave.plugin.enclave.gradle.gramine
+
+import com.r3.conclave.common.internal.PluginUtils.GRAMINE_ENCLAVE_JAR
+import com.r3.conclave.common.internal.PluginUtils.PYTHON_FILE
+import com.r3.conclave.plugin.enclave.gradle.ConclaveTask
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFile
+import org.gradle.process.ExecResult
+import java.io.File
+import javax.inject.Inject
+
+open class GenerateGramineSGXManifest @Inject constructor(objects: ObjectFactory) : ConclaveTask() {
+    companion object {
+        const val GRAMINE_SGX_SIGN_EXECUTABLE = "gramine-sgx-sign"
+        const val GRAMINE_GET_TOKEN_EXECUTABLE = "gramine-sgx-get-token"
+    }
+
+    @get:InputFile
+    val inputPrivateKey: RegularFileProperty = objects.fileProperty()
+
+    @get:InputFile
+    val inputDirectManifest: RegularFileProperty = objects.fileProperty()
+
+    @get:InputFile
+    val inputEnclaveJar: RegularFileProperty = objects.fileProperty()
+
+    @get:InputFile
+    @Optional
+    val inputPythonSourcePath: RegularFileProperty = objects.fileProperty()
+
+    @get:OutputFile
+    val outputSGXManifest: RegularFileProperty = objects.fileProperty()
+
+    @get:OutputFile
+    val outputToken: RegularFileProperty = objects.fileProperty()
+
+    @get:OutputFile
+    val outputSig: RegularFileProperty = objects.fileProperty()
```

Comment:
Let's not follow the pattern of "input"/"output" prefixes. It's unnecessary as the context of the task should make it clear. This is not something common in Gradle plugins I've seen, just an odd decission that was made long ago in this repo. I'm in the process of removing these.